### ISSUE TITLE
When the zipkin host isn't localhost..

### ISSUE THREAD

**Issue by adriancole** [openzipkin/zipkin-ui#25]
from @dragontree101

my zipkin-server is 1.16 and change zipkin-ui in 0.0.1
i use new zipkin-ui, but has error, i need change some ui-config?

i use config is
```
environment=
  KAFKA_ZOOKEEPER="10.215.242.11,10.215.242.12,10.215.242.13:2181/kafka-offline",
  STORAGE_TYPE="elasticsearch",
  SELF_TRACING_ENABLED="true",
  ES_CLUSTER="prod",
  ES_HOSTS="http://10.215.242.15:9200",
  ES_PIPELINE="zipkin",
  KAFKA_STREAMS="4",
  QUERY_PORT="8080"
```

![image](https://cloud.githubusercontent.com/assets/1763935/21539261/22132b2c-cde0-11e6-8217-41b6937a6b4b.png)

i think to change localhost to zipkin-server's ip?  i am not deploy zipkin-server in my local.

**Comment by adriancole** [comment]
looking at the code, it seems there are some constants relating to the fully-qualified api host. is there a way to use a relative path instead? (forgive me, if I misunderstand them)

**Comment by jprateekvmware** [comment]
I am in the same boat , How can I have the localhost replaced with the relative path

**Comment by adriancole** [comment]
this is just waiting on work to do it. does anyone have time to help?

**Comment by ankon** [comment]
What's the general idea for configuring this UI?

I see that [docker-zipkin](https://github.com/openzipkin/docker-zipkin/tree/master/zipkin-ui) uses a nginx to proxy the actual backend onto the same server that hosts the UI, and ZipkinService hardcodes 'localhost' *and* the port here: https://github.com/openzipkin/zipkin-ui/blob/master/src/app/zipkin/zipkin.ts#L155

Seems that one needs to do two things: kill the port reference there, and then fix all places that work with localhost?

**Comment by adriancole** [comment]
@ankon so when in development (and also many sites), the port is 9411. So, I supposed some means of configuring the api endpoint ex (http://localhost:9411) would be what's needed. Then, whatever parameter is needed would be passed to the deployment.

One idea is to expose the api endpoint as a part of config.json, which is how the old UI gets its config, lacking that assume http://localhost:9411. There is a chicken/egg problem, of course as config.json is served by the host listening on 9411!

TL;DR; I don't know the best way, but what you've mentioned makes sense.

cc @openzipkin/devops-tooling

**Comment by adriancole** [comment]
@charlescui made a suggestion here https://github.com/openzipkin/docker-zipkin/issues/134#issuecomment-288403568

**Comment by adriancole** [comment]
here's an example implementation https://github.com/openzipkin/zipkin-ui/pull/34 I think others should comment wrt the configuration approach.

**Comment by adriancole** [comment]
#34 suggests separate env variables to assign this.. like `export ZIPKIN_HOST=172.16.10.56 ZIPKIN_PORT=32008` please give feedback if this works for you or not.
